sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/schema/AvroType.java

```diff
@@ -0,0 +1,366 @@
+package com.azure.storage.internal.avro.implementation.schema;
+
+import com.azure.storage.internal.avro.implementation.AvroConstants;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.ARRAY;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.BOOLEAN;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.BYTES;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.DOUBLE;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.ENUM;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.FIXED;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.FLOAT;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.INT;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.LONG;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.MAP;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.NULL;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.PRIMITIVE_TYPES;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.RECORD;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.STRING;
+import static com.azure.storage.internal.avro.implementation.AvroConstants.Types.UNION;
+
+/**
+ * A class that represents an Avro type.
+ * AvroTypes function as a type that stores all the data a schema may need.
+ * @see AvroType#getType(JsonNode)
+ */
+public class AvroType {
+
+    private final String type;
+
+    /**
+     * Creates a new instance of an AvroType.
+     * @param type The type associated with the AvroType.
+     */
+    public AvroType(String type) {
+        this.type = type;
+    }
+
+    /**
+     * @return the type.
+     */
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * An avro primitive type.
+     * No additional data is required to parse a primitive.
+     */
+    public static class AvroPrimitiveType extends AvroType {
```

Comment:
I think nested classes is not super idiomatic and maybe we should create different independent types for each of these

Reply:
Discussed offline. Nested classes seem ok in this use case as per Oracle docs. 
https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html